www/front_src/src/Resources/Filter/index.tsx

```diff
@@ -52,23 +70,190 @@ const Filter = (): JSX.Element => {
+    autoCompleteSuggestions,
+    open,
+    selectedSuggestionIndex,
     currentFilter,
   ];
 
-  const requestSearchOnEnterKey = (event: React.KeyboardEvent): void => {
-    const enterKeyPressed = event.keyCode === 13;
+  React.useEffect(() => {
+    if (equals(autoCompleteSuggestions, []) || !isSearchFieldFocus) {
+      setAutocompleteAnchor(null);
+
+      return;
+    }
+
+    setAutocompleteAnchor(searchRef?.current as HTMLDivElement);
+  }, [autoCompleteSuggestions]);
+
+  const acceptAutocompleteSuggestionAtIndex = (index: number) => {
+    setNewFilter();
+
+    const acceptedSuggestion = autoCompleteSuggestions[index];
+
+    if (search[cursorPosition] === ',') {
+      setSearch(search + acceptedSuggestion);
+      return;
+    }
+
+    const searchBeforeCursor = search.slice(0, cursorPosition + 1);
+    // the search is composed of "expressions" separated by whitespaces
+    // (like "status:OK" for instance)
+    const expressionBeforeCursor =
+      last(searchBeforeCursor.trim().split(' ')) || '';
+
+    // an expression is "complete" when it has a value that is not in the middle of an input
+    // ("status:"" or "status:OK", for instance, but not "status:O")
+    const isExpressionComplete =
+      expressionBeforeCursor.endsWith(':') ||
+      expressionBeforeCursor.endsWith(',') ||
+      acceptedSuggestion.startsWith(',');
+
+    const expressionAfterSeparator = isExpressionComplete
+      ? ''
+      : last(expressionBeforeCursor.split(/:|,/)) || '';
+
+    const completedWord = acceptedSuggestion.slice(
+      expressionAfterSeparator.length,
+      acceptedSuggestion.length,
+    );
+
+    const cursorCompletionShift =
+      acceptedSuggestion.length - expressionAfterSeparator.length;
+
+    const isExpressionEmpty = expressionAfterSeparator === '';
+    const searchCutPosition = isExpressionEmpty
+      ? cursorPosition + 1
+      : cursorPosition;
+
+    const searchBeforeCompletedWord = search.slice(0, searchCutPosition);
+    const searchAfterCompletedWord = search.slice(searchCutPosition);
+
+    const searchWithAcceptedSuggestion = [
+      searchBeforeCompletedWord.trim(),
+      completedWord,
+      searchAfterCompletedWord.trim() === '' ? '' : ' ',
+      searchAfterCompletedWord,
+    ].join('');
+
+    setCursorPosition(cursorPosition + cursorCompletionShift);
+
+    if (isNil(search[cursorPosition])) {
+      setSearch(searchWithAcceptedSuggestion);
+
+      return;
+    }
+
+    // when the autocompletion takes part somewhere that is not at the end of the output,
+    // we need to shift the correspoding expression to the end, because that's where the cursor will end up
+    const expressionToShiftToTheEnd = expressionBeforeCursor.includes(':')
+      ? expressionBeforeCursor + completedWord
+      : acceptedSuggestion;
+
+    setSearch(
+      `${searchWithAcceptedSuggestion
+        .replace(expressionToShiftToTheEnd, '')
+        .trim()} ${expressionToShiftToTheEnd}`.trim(),
+    );
```

Comment:
To make it a bit less confusing, I propose something like this. It could be good to refactor a bit to make it nice among the code
```javascript
[
  searchAfterCompletedWord.trim(),
  ' ',
  searchBeforeCompletedWord.trim(),
  completedWord,
].join('')
```

Reply:
I'll look into that